src/core.ts

```diff
@@ -550,15 +551,45 @@ export class AttributePart implements Part {
 
 export class NodePart implements Part {
   templateFactory: TemplateFactory;
-  startNode: Node;
-  endNode: Node;
+  startNode!: Node;
+  endNode!: Node;
   _value: any = undefined;
 
-  constructor(
-      startNode: Node, endNode: Node, templateFactory: TemplateFactory) {
+  constructor(templateFactory: TemplateFactory) {
     this.templateFactory = templateFactory;
-    this.startNode = startNode;
-    this.endNode = endNode;
+  }
+
+  /**
+   * Inserts this part between `ref` and `ref`'s next sibling. Both `ref` and
+   * its next sibling must be static, unchanging nodes such as those that appear
+   * in a literal section of a template.
+   *
+   * This part must be empty, as its contents are not automatically moved.
+   */
+  insertAfterNode(ref: Node) {
+    this.startNode = ref;
+    this.endNode = ref.nextSibling!;
+  }
+
+  /**
+   * Appends this part into a parent part.
+   *
+   * This part must be empty, as its contents are not automatically moved.
+   */
+  appendIntoPart(part: NodePart) {
```

Comment:
Definitely not a slam dunk, but these could be more like their element counterparts as `appendPart`, `insertPartBefore`... wdyt?

Reply:
I went around on the names a bit and I still think these are the most clear.